sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -14,56 +13,48 @@ use azure_core::{
 };
 use azure_identity::DefaultAzureCredentialBuilder;
 use std::collections::HashMap;
-use std::marker::PhantomData;
 use std::sync::Arc;
 use uuid::Uuid;
 
-pub struct BlobClient<BlobType = Unset> {
-    pub(crate) blob_type: PhantomData<BlobType>,
+pub struct BlobClient {
     pub(crate) endpoint: String,
     pub(crate) container_name: String,
     pub(crate) blob_name: String,
-    pub(crate) credential: Option<Arc<dyn TokenCredential>>,
+    pub(crate) credential: Arc<dyn TokenCredential>,
     pub(crate) client: GeneratedBlobClient,
 }
 
-impl BlobClient<Unset> {
+impl BlobClient {
     pub fn new(
         endpoint: String,
         container_name: String,
         blob_name: String,
-        credential: Option<Arc<dyn TokenCredential>>,
-        mut options: Option<BlobClientOptions>,
+        credential: Arc<dyn TokenCredential>,
+        mut options: BlobClientOptions,
```

Comment:
Keep this as optional

Reply:
Why is this `mut`? That burdens callers unnecessarily. Per the guidelines, this must be an `Option<BlobClientOptions>` and in your body you do something like:

```rust
let options = options.unwrap_or_default();
```

If options aren't optional, they shouldn't be in your client *options*.